reports.py

```diff
@@ -673,6 +680,22 @@ def plot_coverage(
             segment_depths.setdefault(row[0], []).append(float(row[2]))
             domain_max += 1
 
+    # Binning
+    bin_size = 1
+    if bin_large_plots:
+        # Number points plotted after binning is set to true width of plot (excluding
+        # axis titles and labels and whitespace around plot), calculated by trial and error
+        # There is one bin per pixel
+        preferred_domain = int(0.907634 * plot_width - 103.207)
+
+        # Bin locations and take maximum in each bin
+        bin_size = 1 + int(domain_max/preferred_domain)
+        binned_segment_depths = OrderedDict()
+        for segment_num, (segment_name, position_depths) in enumerate(segment_depths.items()):
+            max_depths_in_bins = [max(position_depths[i:i + bin_size]) for i in range(0, len(position_depths), bin_size)]
```

Comment:
This may be a question for others who routinely look at these plots, but since these plots are often used to spot regions of low coverage, I wonder if the `max` depth of positions within a bin is the stat we care about most, or if `min` may be more appropriate. Or maybe the value could be the number of positions within a bin exceeding some threshold value (the mean coverage within the bin?)? Perhaps it should be user-configurable?

Reply:
Sure. When I was playing around with different options, I found that the max best represented the actual shape of the plot, which is what I wanted to see. I have also been more interested in the presence rather than absence of coverage. I don't see any harm in giving people options—but I would want max to be one of those options, since it has served me quite well.